src/algebra/monoid_algebra/basic.lean

```diff
@@ -244,11 +244,14 @@ instance [ring k] [semigroup G] : non_unital_ring (monoid_algebra k G) :=
   .. monoid_algebra.non_unital_semiring }
 
 instance [ring k] [mul_one_class G] : non_assoc_ring (monoid_algebra k G) :=
-{ .. monoid_algebra.add_comm_group,
+{ int_cast                    := λ n, single 1 (algebra_map ℤ k n),
```

Comment:
I think this might be better as
```suggestion
{ int_cast                    := λ z, single 1 (z : k),
```
Just to make it clear this is just applying a lower-level int cast